src/python/lib/clusterfuzz/stacktraces/__init__.py

```diff
@@ -0,0 +1,1362 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Stack parsing module."""
+import os
+import re
+import string
+import subprocess
+
+from base import utils
+from crash_analysis import crash_analyzer
+from metrics import logs
+from system import environment
+
+from .constants import *
+
+
+class CrashInfo:
+  """Parsed crash information."""
+
+  def __init__(self):
+    self.crash_type = ''
+    self.crash_address = ''
+    self.crash_state = ''
+    self.crash_stacktrace = ''
+    self.frame_count = 0
+    self.process_name = 'NULL'
+    self.process_died = False
+
+    # Following fields are for internal use only and subject to change. Do not
+    # rely on these.
+    self.frames = []
+    self.last_frame_id = -1
+    self.raw_frames = []
+
+    # Additional tracking for Android bugs.
+    self.found_java_exception = False
+
+    # Additional tracking for bad casts.
+    self.found_bad_cast_crash_end_marker = False
+
+    # Additional tracking for check failures.
+    self.check_failure_source_file = ''
+
+    # Additional tracking for fatal errors.
+    self.fatal_error_occurred = False
+
+    self.is_kasan = False
+    self.is_golang = False
+    self.is_python = False
+    self.found_python_crash = False
+    self.found_golang_crash = False
+    self.found_android_kernel_crash = False
+
+
+def _filter_stack_frame(stack_frame):
```

Comment:
nit: can be moved after StackParser, next to other funvtions

Reply:
This can't be moved as it has to come before the default argument value in add_frame_on_match.